### DISCUSSION TITLE
Suggestion: spin-off the `distributions` module into a shared common dependency among all *lss modules

### DISCUSSION THREAD

**Discussion by gmgeorg** [StatMixedML/XGBoostLSS#67] [ToDo]
First of all thank you for working on several boosting LSS versions (xgboost, catboost, lightgbm)!

I did notice that both xgb and lightgbm have the (at first sight) exact same `distributions.py` submodule that is the `torch` core of for *any* distribution supported as a boosting prediction:

* https://github.com/StatMixedML/XGBoostLSS/tree/master/xgboostlss/distributions
* https://github.com/StatMixedML/LightGBMLSS/tree/master/lightgbmlss/distributions

Is there a specific reason to not share the distribution modules (and any other shared functionality, like plotting of distribution results) in a common Python module that then gets imported/shared as a common dependency here to avoid duplication; and separate development from adding distributions vs adding functionality to the boosting part of the modules.

**Comment by StatMixedML** [comment]
> Is there a specific reason to not share the distribution modules 

Not specifically, except to have the distributions explicit in each framework. You are free to suggest a different way. Thanks.
